### PR TITLE
Switch to poetry, drop support for Python 3.5 and Django 2.1

### PR DESCRIPTION
**Pull request by dyve**
Uses poetry for packaging. Removes the need for setup.py, setuptools, setuptools_scm and lots more.

Upon installing poetry, I found that black is not compatible with python 3.5. Instead of finding a fix, I removed Python 3.5 support. It's EOL 2020-09-13, and removing 3.5 support will enable f-strings for this package.

Extended support for Django 2.1 ended in December 2, 2019, so we also dropped support for this version.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tox]
+isolated_build = true
 envlist =
-    {py35}-{django21,django22}
     {py36}-{django21,django22,django30,djangomaster}
```
[the remark this answers is not in the document]
Comment (dyve) [reply tox.ini]: Removing 2.1 and clarifying in README.

File: src/django_icons/__init__.py [hunk 2]
```diff
@@ -1,16 +1,5 @@
...
-    from ._version import version
-except ImportError:
-    try:
-        from setuptools_scm import get_version
-
-        version = get_version()
-    except ImportError:
-        version = "???"
-__version__ = version
-
```
[the remark this answers is not in the document]
Comment (dyve) [reply src/django_icons/__init__.py]: Yes, apparently poetry does not support that. But I added it again using a different route.

File: CHANGELOG.rst [hunk 3]
```diff
@@ -1,6 +1,12 @@
 Changelog
 ---------
 
+In development
+==============
+
+- Drop support for Python 3.5
+- Switch to poetry
```
[the remark this answers is not in the document]
Comment (dyve) [reply CHANGELOG.rst]: Agreed and fixed.

**Comment by dyve** [comment]
OK to merge this?

**Comment by jieter** [comment]
looks good to me

**Pull request merged**